stock_split_picking_kit/wizards/stock_split_picking.py

```diff
@@ -0,0 +1,115 @@
+    _inherit = "stock.split.picking"
+
+    mode = fields.Selection(
+        selection_add=[("kit_quantity", "Quantity of kits")],
+        ondelete={"kit_quantity": "set default"},
+    )
+    kit_split_quantity = fields.Integer(string="Number of kits by transfer")
+
+    def _apply_kit_quantity(self):
+        new_pickings = self.env["stock.picking"]
+        for picking in self.picking_ids:
+            new_moves_vals = []
+            moves_to_split_off = self.env["stock.move"]
+            # Do not split off moves that are done or cancelled
+            todo_qty = self.kit_split_quantity
+            todo_moves = picking.move_ids.filtered(
+                lambda m: m.state not in ("done", "cancel")
+            ).sorted()
+            todo_moves_by_bom = todo_moves.grouped(lambda m: m.bom_line_id.bom_id)
+            # Process each bom
+            for bom, moves in todo_moves_by_bom.items():
+                # Stop processing if there's nothing else to split off
+                if todo_qty <= 0:
+                    break
+                # Handle kits
+                if bom.type == "phantom":
+                    kit_quantity = moves._compute_kit_quantities(
+                        bom.product_id,
+                        bom.product_qty,
+                        bom,
+                        {
+                            "incoming_moves": lambda m: True,
+                            "outgoing_moves": lambda m: False,
+                        },
+                    )
+                    # If the kit quantity is lower or equal than the todo_qty, we can
+                    # split off the whole kit moves without any move splitting.
+                    if (
+                        float_compare(
+                            kit_quantity,
+                            todo_qty,
+                            precision_rounding=bom.product_uom_id.rounding,
+                        )
+                        <= 0
+                    ):
+                        todo_qty -= kit_quantity
+                        moves_to_split_off += moves
+                        continue
+                    # Otherwise, we'd be consuming the complete todo_qty, but we need to
+                    # split the kit component moves.
+                    for move in moves:
+                        new_moves_vals += move.with_context(
+                            cancel_backorder=False
+                        )._split(
+                            move.product_uom._compute_quantity(
+                                move.bom_line_id.product_qty * todo_qty,
+                                move.product_id.uom_id,
+                                rounding_method="HALF-UP",
+                            )
+                        )
+                    # If we got this far, we've consumed all the todo_qty
+                    todo_qty = 0
+                    break
+                # Handle regular products: their quantity is the number of slots used
```

Comment:
as you drop "slot" term everywhere
```suggestion
                # Handle regular products: use their quantity
```